File: YYDroid/src/main/scala/me/reyoung/YYDroid/util/DatabaseUtil.scala
package me.reyoung.YYDroid.util

import android.content.Context
import com.j256.ormlite.android.apptools.OrmLiteSqliteOpenHelper
import android.database.sqlite.SQLiteDatabase
import com.j256.ormlite.support.ConnectionSource
import com.j256.ormlite.table.TableUtils
import me.reyoung.YYDroid.model.Subscribe
import android.util.Log
import java.sql.SQLException

/**
 * Created with IntelliJ IDEA.
 * User: reyoung
 * Date: 2/16/13
 * Time: 5:51 PM
 * To change this template use File | Settings | File Templates.
 */
object DatabaseUtil{
  val DATABASE_NAME= "YYDroid.db"
  val DATABASE_VERSION=0
  val TABLES = Array(classOf[Subscribe])
}

class DatabaseUtil(val context:Context) extends OrmLiteSqliteOpenHelper(
  context,DatabaseUtil.DATABASE_NAME,null,DatabaseUtil.DATABASE_VERSION
) with LogTag {
  def onCreate(db: SQLiteDatabase, cs: ConnectionSource) {
    try{
      Log.d(LogTag,"Creating Tables")
      DatabaseUtil.TABLES.foreach(cls=>TableUtils.createTable(cs,cls))
    } catch {
      case ex:SQLException =>{
        Log.e(LogTag, "Unable to create databases",ex)
      }
    }
  }

  def onUpgrade(db: SQLiteDatabase, cs: ConnectionSource, oldVer: Int, newVer: Int) {
    try {
      DatabaseUtil.TABLES.reverse.foreach(cls=>TableUtils.dropTable(cs,cls,true))
      this.onCreate(db,cs)
    } catch {
      case ex:SQLException => {
        Log.e(LogTag,"Unable to update databases",ex)
      }
    }
  }
}

File: YYDroid/src/main/java/me/reyoung/YYDroid/util/DatabaseUtil.java
package me.reyoung.YYDroid.util;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.util.Log;
import com.j256.ormlite.android.apptools.OrmLiteSqliteOpenHelper;
import com.j256.ormlite.dao.Dao;
import com.j256.ormlite.support.ConnectionSource;
import com.j256.ormlite.table.TableUtils;
import me.reyoung.YYDroid.model.Subscriber;

import java.sql.SQLException;

public class DatabaseUtil extends OrmLiteSqliteOpenHelper {
    static final private String DATABASE_NAME= "YYDroid.db";
    static final private int DATABASE_VERSION=2;
    static final private Class TABLES[] = {
            Subscriber.class
    };
    private Dao<Subscriber, Integer> subscriberDao=null;

    public DatabaseUtil(Context ctx){
        super(ctx,DATABASE_NAME,null,DATABASE_VERSION);
    }
    @Override
    public void onCreate(SQLiteDatabase sqLiteDatabase, ConnectionSource connectionSource) {
        try {
            Log.d(LogTag(),"Create Tables");
            for (Class cls :DatabaseUtil.TABLES){
                TableUtils.createTable(connectionSource,cls);
            }
        } catch (SQLException ex){
            Log.e(LogTag(),"Unable to create database",ex);
        }
    }

    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, ConnectionSource connectionSource, int i, int i2) {
        try {
            Log.d(LogTag(),"Update Table");
            for (int k=0;i<TABLES.length;++k){
                TableUtils.dropTable(connectionSource,TABLES[TABLES.length-k],true);
            }
        } catch (SQLException e) {
            Log.e(LogTag(),"Unable to update database",e);
        }
    }

    public String LogTag() {
        return "YYDroid_"+this.getClass().getSimpleName();
    }

    public Dao<Subscriber, Integer> getSubscriberDao() throws SQLException {
        if(subscriberDao==null){
            subscriberDao = getDao(Subscriber.class);
        }
        return subscriberDao;
    }
}